src/core/api/chart_repository.go

```diff
@@ -355,13 +384,32 @@ func (cra *ChartRepositoryAPI) DeleteChart() {
 		return
 	}
 
+	versions := []string{}
 	for _, chartVersion := range chartVersions {
+		versions = append(versions, chartVersion.GetVersion())
 		if err := cra.removeLabelsFromChart(chartName, chartVersion.GetVersion()); err != nil {
 			cra.SendInternalServerError(err)
 			return
 		}
 	}
 
+	event := &n_event.Event{}
+	metaData := &n_event.ChartDeleteMetaData{
+		ChartMetaData: n_event.ChartMetaData{
+			ProjectName: cra.namespace,
+			ChartName:   chartName,
+			Versions:    versions,
+			OccurAt:     time.Now(),
+			Operator:    cra.SecurityCtx.GetUsername(),
+		},
+	}
+	if err := event.Build(metaData); err != nil {
+		hlog.Errorf("failed to build chart delete event metadata: %v", err)
+	}
+	if err := event.Publish(); err != nil {
```

Comment:
delete chart event should be published after `chartController.DeleteChart`?

Reply:
you are right. but it will be triggered regardless of the result of DeleteChart for the action is not atomic and some charts may have been deleted even error returned.